backend-shared/plannershared.cpp

```diff
@@ -211,3 +212,15 @@ void plannerShared::set_sacfactor(double value)
 	// NO conversion, this is done in the planner model.
 	DivePlannerPointsModel::instance()->setSacFactor(value);
 }
+
+bool plannerShared::o2narcotic()
+{
+	return qPrefDivePlanner::o2narcotic();
+}
+void plannerShared::set_o2narcotic(bool value)
+{
+	qPrefDivePlanner::set_o2narcotic(value);
+	DivePlannerPointsModel::instance()->emitDataChanged();
+	CylindersModel::instance()->updateBestMixes();
```

Comment:
just reading this... we call a setter in `plannerShared` which calls a setter in `qPrefDivePlanner`, emits a signal from `DivePlannerPointsModel` and then calls a random update function in `CylinderModel`.

That looks like a complete mess, not like a clean design (and use) of those classes.
Anyone trying to debug this in a year will be thoroughly confused, I think.